src/lib/crypto/kimchi_backend/gadgets/ecdsa.ml

```diff
@@ -404,12 +411,13 @@ let verify (type f) (module Circuit : Snark_intf.Run with type field = f)
    *           Result bound check is already covered by scalar_checks
    *)
   Foreign_field.External_checks.append_bound_check scalar_checks quotient ;
+  Foreign_field.External_checks.append_canonical_check scalar_checks quotient ;
```

Comment:
Pending task for myself: check more carefully how many of these canonical checks we need